testcommon/mockSmartContractCallerTest.go

```diff
@@ -0,0 +1,131 @@
+func CreateMockContract(address []byte) *MockTestSmartContract {
+	return &MockTestSmartContract{
+		testSmartContract: testSmartContract{
+			address: address,
+		},
+	}
+}
+
+// WithBalance provides the balance for the MockTestSmartContract
+func (mockSC *MockTestSmartContract) WithBalance(balance int64) *MockTestSmartContract {
+	mockSC.balance = balance
+	return mockSC
+}
+
+// WithConfig provides the config object for the MockTestSmartContract
+func (mockSC *MockTestSmartContract) WithConfig(config interface{}) *MockTestSmartContract {
+	mockSC.config = config
+	return mockSC
+}
+
+// WithMethods provides the methods for the MockTestSmartContract
+func (mockSC *MockTestSmartContract) WithMethods(initMethods ...func(*mock.InstanceMock, interface{})) MockTestSmartContract {
+	mockSC.initMethods = initMethods
+	return *mockSC
+}
+
+func (mockSC *MockTestSmartContract) initialize(t testing.TB, host arwen.VMHost, imb *mock.InstanceBuilderMock) {
+	instance := imb.CreateAndStoreInstanceMock(t, host, mockSC.address, mockSC.balance)
+	for _, initMethod := range mockSC.initMethods {
+		initMethod(instance, mockSC.config)
+	}
+}
+
+// MockInstancesTestTemplate holds the data to build a mock contract call test
+type MockInstancesTestTemplate struct {
+	testTemplateConfig
+	contracts     *[]MockTestSmartContract
+	setup         func(arwen.VMHost, *worldmock.MockWorld)
+	assertResults func(*worldmock.MockWorld, *VMOutputVerifier)
+}
+
+// BuildMockInstanceCallTest starts the building process for a mock contract call test
+func BuildMockInstanceCallTest(t *testing.T) *MockInstancesTestTemplate {
+	return &MockInstancesTestTemplate{
+		testTemplateConfig: testTemplateConfig{
+			t:        t,
+			useMocks: true,
+		},
+		setup: func(arwen.VMHost, *worldmock.MockWorld) {},
+	}
+}
+
+// WithContracts provides the contracts to be used by the mock contract call test
+func (callerTest *MockInstancesTestTemplate) WithContracts(usedContracts ...MockTestSmartContract) *MockInstancesTestTemplate {
+	callerTest.contracts = &usedContracts
+	return callerTest
+}
+
+// WithInput provides the ContractCallInput to be used by the mock contract call test
+func (callerTest *MockInstancesTestTemplate) WithInput(input *vmcommon.ContractCallInput) *MockInstancesTestTemplate {
+	callerTest.input = input
+	return callerTest
+}
+
+// WithSetup provides the setup function to be used by the mock contract call test
+func (callerTest *MockInstancesTestTemplate) WithSetup(setup func(arwen.VMHost, *worldmock.MockWorld)) *MockInstancesTestTemplate {
+	callerTest.setup = setup
+	return callerTest
+}
+
+// AndAssertResults provides the function that will aserts the results
+func (callerTest *MockInstancesTestTemplate) AndAssertResults(assertResults func(world *worldmock.MockWorld, verify *VMOutputVerifier)) {
+	callerTest.assertResults = assertResults
+	callerTest.runTest()
+}
+
+func (callerTest *MockInstancesTestTemplate) runTest() {
+
```

Comment:
To find all blank lines at the top of functions, run this in a terminal in the root of `arwen-wasm-vm`:
`grep -PzRo 'func .*{\n\n' *`

Reply:
done